event/src/main/java/io/airlift/event/client/EventFieldMetadata.java

```diff
@@ -71,6 +73,21 @@ public String toString()
         this.containerType = containerType;
     }
 
+    public String getName()
+    {
+        return name;
+    }
+
+    public EventTypeMetadata<?> getNestedType()
```

Comment:
I'd make this return Optional, and add a comment in the javadoc explaining that it's "empty()" for atomic types.